streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindows.java

```diff
@@ -125,7 +125,11 @@ public static TimeWindows of(final long sizeMs) throws IllegalArgumentException
      * @throws IllegalArgumentException if the specified window size is zero or negative or can't be represented as {@code long milliseconds}
      */
     public static TimeWindows of(final Duration size) throws IllegalArgumentException {
-        ApiUtils.validateMillisecondDuration(size, "size");
+        try {
+            ApiUtils.validateMillisecondDuration(size, "size");
+        } catch (final IllegalArgumentException e) {
+            throw new IllegalArgumentException("The size of the window is incorrect, value : " + size, e);
```

Comment:
I felt the same, but I picked this approach to make less changes in the code, since I am beginner here.

Regarding `validateMillisecondDuration()` update:  
Let me bring up another thought!!! 
> I'd feel `throw`ing appropriate custom/existing exceptions(like `NullArgumentException`,  `ArithmeticException/NumberOverflowException`) with valid message in exception object as per failed cases make easy to the callers(like `advanceBy()`) to know what's kind of check failed.  
-- thoughts ??

Adding a prefix for error message is also good idea. But, I need to give a thought on kind of prefixes we should use to make it unique for various fail cases.

Reply:
> I felt the same, but I picked this approach to make less changes in the code, since I am beginner here.

That's fair. However, we should not "optimize" for fewer changes but for better code. :)

> But, I need to give a thought on kind of prefixes we should use to make it unique for various fail cases.

`validateMillisecondDuration()` will just add the prefix. Each called can pass in whatever prefix is suitable for it.

Reply:
Any thought about how the prefix text should look like?

Reply:
That's also fine :) !!!

How about using the second param as `prefix`, since we are using second param only for name in error message preparation.

Thinking of something like this :

    public static long validateMillisecondDuration(final Duration duration, final String message) {
        try {
            if (duration == null)
                throw new IllegalArgumentException(message);

            return duration.toMillis();
        } catch (final ArithmeticException e) {
            throw new IllegalArgumentException(message, e);
        }
    } 

One of the caller will be:-

    # in TimeWindows#advanceBy(final Duration advance)
    ApiUtils.validateMillisecondDuration(advance, "The advance interval for TimeWindows is incorrect, value : " + advance);

Generalized caller version:-

    ApiUtils.validateMillisecondDuration(advance, "The {variable name} for {class name} is incorrect, value : " + {value param});

Reply:
SGTM.

Reply:
Perfect!! committed code changes.